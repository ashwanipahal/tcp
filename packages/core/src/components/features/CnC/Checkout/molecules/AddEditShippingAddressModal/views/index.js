export { default } from './AddEditShippingAddressModal.view';
