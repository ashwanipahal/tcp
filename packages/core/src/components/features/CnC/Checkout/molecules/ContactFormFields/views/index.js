export { default } from './ContactFormFields.view';
