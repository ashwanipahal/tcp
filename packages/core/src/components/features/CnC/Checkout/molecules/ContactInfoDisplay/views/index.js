export { default } from './ContactInfoDisplay.view';
