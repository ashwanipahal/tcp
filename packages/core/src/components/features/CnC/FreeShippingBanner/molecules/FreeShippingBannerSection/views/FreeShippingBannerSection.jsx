import React from 'react';
import { PropTypes } from 'prop-types';
import { getLabelValue } from '@tcp/core/src/utils';
import withStyles from '../../../../../../common/hoc/withStyles';
import Styles from '../styles/FreeShippingBannerSection.style';
import { BodyCopy, Anchor, Image } from '../../../../../../common/atoms';
import { labelsHashValuesReplace, convertHtml } from '../../../../LoyaltyBanner/util/utility';
import CONSTANTS from '../../../../Checkout/Checkout.constants';
import { getIconPath } from '../../../../../../../utils';

const FreeShippingBannerSection = props => {
  const fastShippingPath = getIconPath('fast-shipping');
  const { className, labels } = props;

  const utilArrShippingNew = [
    {
      key: '#tagOpen#',
      value: `<span class="${className} shippingNew">`,
    },
    {
      key: '#tagClose#',
      value: `</span>`,
    },
  ];
  const labelStr = getLabelValue(labels, 'lbl_freeShippingBanner_label');
  const convertedStrValue = labelsHashValuesReplace(labelStr, utilArrShippingNew);
  const freeShippingLabel = convertHtml(convertedStrValue);
  const freeShippingLabelExist = freeShippingLabel !== 'lbl_freeShippingBanner_label';

  return (
    freeShippingLabelExist && (
      <div className={`${className}`}>
        <BodyCopy
          className="free-shipping-banner-section-wrapper"
          component="div"
          fontFamily="secondary"
          fontWeight="extrabold"
        >
          <Image src={fastShippingPath} alt="Fast Shipping" className="fast-shipping" />
          {freeShippingLabel}
          <span className="free-shipping-details">
            <Anchor
              className="details"
              anchorVariation="primary"
              text={getLabelValue(labels, 'lbl_freeShippingBanner_details')}
              url={CONSTANTS.FREE_SHIPPING_URL}
              target="_blank"
              underline
            />
          </span>
        </BodyCopy>
      </div>
    )
  );
};

FreeShippingBannerSection.propTypes = {
  className: PropTypes.string,
  labels: PropTypes.shape({}).isRequired,
};

FreeShippingBannerSection.defaultProps = {
  className: '',
};

export default withStyles(FreeShippingBannerSection, Styles);
export { FreeShippingBannerSection as FreeShippingBannerSectionVanilla };
