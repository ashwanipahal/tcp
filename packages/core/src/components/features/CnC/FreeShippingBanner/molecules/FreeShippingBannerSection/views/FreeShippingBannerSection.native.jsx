import React from 'react';
import { PropTypes } from 'prop-types';
import { View } from 'react-native';
import { getLabelValue } from '@tcp/core/src/utils';
import {
  FreeShippingLabel,
  AnchorWrapper,
  FreeShippingIconWrapper,
} from '../styles/FreeShippingBannerSection.style.native';
import { BodyCopy, Anchor, Image } from '../../../../../../common/atoms';
import mobileHashValues from '../../../../LoyaltyBanner/util/utilityNative';
import fastShipping from '../../../../../../../assets/fast-shipping.png';

const FreeShippingBannerSection = props => {
  const { labels } = props;
  const utilArrShippingNew = [
    {
      key: '#tagOpen# ',
      value: 'ShippingNew',
    },
  ];
  const labelStr = getLabelValue(labels, 'lbl_freeShippingBanner_label');
  const freeShippingURL = getLabelValue(labels, 'lbl_freeShippingBanner_url_app');
  const freeShippingLabel = labelStr ? mobileHashValues(labelStr, utilArrShippingNew) : false;

  return (
    <>
      {freeShippingLabel && (
        <FreeShippingLabel>
          <FreeShippingIconWrapper>
            <Image
              source={fastShipping}
              alt={getLabelValue(labels, 'lbl_fast_shipping')}
              height={18}
              width={20}
            />
          </FreeShippingIconWrapper>
          <View>
            <BodyCopy
              mobilefontFamily={['secondary']}
              textAlign="center"
              fontWeight="extrabold"
              text={freeShippingLabel}
            />
          </View>
          <AnchorWrapper>
            <Anchor
              className="details"
              anchorVariation="primary"
              text={getLabelValue(labels, 'lbl_freeShippingBanner_details')}
              underline
              url={freeShippingURL}
            />
          </AnchorWrapper>
        </FreeShippingLabel>
      )}
    </>
  );
};

FreeShippingBannerSection.propTypes = {
  labels: PropTypes.shape.isRequired,
};

export default FreeShippingBannerSection;
