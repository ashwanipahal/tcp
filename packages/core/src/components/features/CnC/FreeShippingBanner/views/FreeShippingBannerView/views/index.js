export { default } from './FreeShippingBanner.view';
