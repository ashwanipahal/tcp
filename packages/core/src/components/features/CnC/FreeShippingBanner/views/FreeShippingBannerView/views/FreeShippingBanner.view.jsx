import React from 'react';
import PropTypes from 'prop-types';
import FreeShippingBannerSection from '../../../molecules/FreeShippingBannerSection';

const FreeShippingBanner = props => {
  const { className, labels } = props;
  return (
    <div className={className}>
      <FreeShippingBannerSection labels={labels} />
    </div>
  );
};

FreeShippingBanner.propTypes = {
  className: PropTypes.string,
  labels: PropTypes.shape({}).isRequired,
};

FreeShippingBanner.defaultProps = {
  className: '',
};

export default FreeShippingBanner;
