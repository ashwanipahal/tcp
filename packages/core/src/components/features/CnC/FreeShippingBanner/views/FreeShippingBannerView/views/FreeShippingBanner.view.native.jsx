import React from 'react';
import PropTypes from 'prop-types';
import FreeShippingBannerSection from '../../../molecules/FreeShippingBannerSection';

const FreeShippingBanner = props => {
  const { labels } = props;
  return <FreeShippingBannerSection labels={labels} />;
};

FreeShippingBanner.propTypes = {
  labels: PropTypes.shape({}).isRequired,
};

export default FreeShippingBanner;
