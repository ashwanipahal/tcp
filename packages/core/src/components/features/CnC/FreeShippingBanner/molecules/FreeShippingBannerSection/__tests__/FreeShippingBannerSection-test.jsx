import React from 'react';
import { shallow } from 'enzyme';
import { FreeShippingBannerSectionVanilla } from '../views/FreeShippingBannerSection';

describe('FreeShippingBannerSection View Component', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      className: '',
      labels: {},
    };
  });

  it('FreeShippingBannerSection should render correctly', () => {
    component = shallow(<FreeShippingBannerSectionVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('FreeShippingBannerSection with label should render correctly', () => {
    props.labels = 'abc';
    component = shallow(<FreeShippingBannerSectionVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
});
