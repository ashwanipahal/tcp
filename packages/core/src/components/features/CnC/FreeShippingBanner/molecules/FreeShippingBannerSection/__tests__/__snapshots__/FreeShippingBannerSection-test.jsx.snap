// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FreeShippingBannerSection View Component FreeShippingBannerSection should render correctly 1`] = `
<div
  className=""
>
  <Styled(BodyCopy)
    className="free-shipping-banner-section-wrapper"
    color="text.primary"
    component="div"
    fontFamily="secondary"
    fontSize={12}
    fontWeight="extrabold"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    textAlign="left"
  >
    <Styled(Image)
      className="fast-shipping"
      src="/static/images/fast-shipping.svg"
    />
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "lbl_freeShippingBanner_label",
        }
      }
    />
    <span
      className="free-shipping-details"
    >
      <Styled(Anchor)
        anchorVariation="primary"
        className="details"
        target="_blank"
        text="lbl_freeShippingBanner_details"
        underline={true}
        url="/content/free-shipping"
      />
    </span>
  </Styled(BodyCopy)>
</div>
`;

exports[`FreeShippingBannerSection View Component FreeShippingBannerSection with label should render correctly 1`] = `
<div
  className=""
>
  <Styled(BodyCopy)
    className="free-shipping-banner-section-wrapper"
    color="text.primary"
    component="div"
    fontFamily="secondary"
    fontSize={12}
    fontWeight="extrabold"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    textAlign="left"
  >
    <Styled(Image)
      className="fast-shipping"
      src="/static/images/fast-shipping.svg"
    />
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "lbl_freeShippingBanner_label",
        }
      }
    />
    <span
      className="free-shipping-details"
    >
      <Styled(Anchor)
        anchorVariation="primary"
        className="details"
        target="_blank"
        text="lbl_freeShippingBanner_details"
        underline={true}
        url="/content/free-shipping"
      />
    </span>
  </Styled(BodyCopy)>
</div>
`;
